### PR TITLE
Implementing From UUID trait for GUID

### PR DESCRIPTION
**Pull request by dawid-nowak**
Small addition to enable something like this
``` Rust
let uuid = Uuid::parse_str("1a1a1a1a-2b2b-3c3c-4142-434546474849").unwrap();
let guid: GUID = uuid.into();
```

### CODE REVIEW

File: win_etw_provider/Cargo.toml [hunk 1]
```diff
@@ -14,13 +14,16 @@ readme = "../README.md"
 widestring = "^1.0"
 zerocopy = "^0.6"
 win_etw_metadata = { version = "^0.1.2", path = "../win_etw_metadata" }
+uuid = "1"
```
Comment (sivadeilra) [inline win_etw_provider/Cargo.toml:17]: Can you make this an optional dependency / feature?

**Comment by sivadeilra** [comment]
Thanks for the contribution!  Just one quick request, to make the `uuid` dependency optional.

File: win_etw_provider/Cargo.toml (hunk 1 above)
Comment (dawid-nowak) [reply win_etw_provider/Cargo.toml]: done

File: win_etw_provider/Cargo.toml [hunk 2]
```diff
@@ -28,5 +28,8 @@ uuid="1"
 [features]
 std = []
 default = ["std"]
+uuid =["dep:uuid"]
```
Comment (sivadeilra) [inline win_etw_provider/Cargo.toml:31]: Is this the "new" way to do optional dependencies?  I've never seen this syntax before, and I don't see it in the TOML manifest reference.  I'm happy to take the change, this is just the first I've seen of this.

Comment (dawid-nowak) [reply win_etw_provider/Cargo.toml:31]: yes, it seems to be https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies

**Comment by sivadeilra** [comment]
Merged, versions bumped, and re-published.  Thanks!

**Pull request merged**